devito/interfaces.py

```diff
@@ -24,6 +24,16 @@
 # manipulation with the correct shapes, pointers, etc.
 _SymbolCache = {}
 
+# This global stores whether newly allocated memory should be initialised
+# using Devito's own JIT engine. It will otherwise be initialised using
+# numpy.
+Global_First_Touch = None
+
+def set_global_first_touch(flag):
```

Comment:
I don't think this has anything to do with convention. The callback mechanism for the `log_level` is required to update the logger object if the value is changed at runtime. Since the FT variable is completely read-only and we're not "undoing" first touch on anything previously allocated, I don't see the need for a callback mechanism here. The only thing it does is some minor type-checking; is this really required here, or should this be part of the `Parameter` class?